integration-tests/utils/integration-utils.js

```diff
@@ -119,6 +123,15 @@ const waitForNoSpinners = page => {
   return page.waitForXPath('//*[@data-icon="loadingSpinner"]', { hidden: true })
 }
 
+// waitForNoSpinnersAfterAction
+// This takes advantage of the mutationObserver in puppeteer by initializing the wait before taking an action.
+// This helps us guarantee that we will catch the spinner before it disappears
+const waitForNoSpinnersAfterAction = async (page, action) => {
+  const foundSpinner = page.waitForXPath('//*[@data-icon="loadingSpinner"]')
+  await Promise.all([foundSpinner, action()])
+  return page.waitForXPath('//*[@data-icon="loadingSpinner"]', { hidden: true })
```

Comment:
Should we factor out the common code?

```suggestion
  return waitForNoSpinners(page)
```